### PR TITLE
Dock/Undock behaviour to 3D views

### LINKED ISSUE #7279: Implement 3D map viewers manager (Request in QGIS)
### Request for documentation
From pull request QGIS/qgis#46659
Author: @NEDJIMAbelgacem
QGIS version: 3.24

**Implement 3D map viewers manager**

## Description
Previously when a user closes a 3D map viewer then saves the project, the 3D map viewer will be lost.
This PR adds a 3D map views manager that takes care of removing 3D map views from the project and also renaming and duplication of 3D map views.

- 3D Map Views menu:

Previously, QGIS app view menu contained only "New 3D Map View" action. The PR replaces that action with "3D Map Views" sub menu, the sub menu contains the created map views for easy access, the "New 3D Map View" action and "Manage 3D Map Views". Screenshot of the "3D Map Views" menu :

Screenshot from 2021-12-30 12-37-48

- 3D Map Views Manager dialog:

The 3D Map Views Manager gives users the ability to open, duplicate, rename and remove 3D map views, Screenshot for the 3D map views manger:

Screenshot from 2021-12-30 12-42-48
The rename and duplicate buttons will ask the user for a new 3D Map viewer title.

### Commits tagged with [need-docs] or [FEATURE]

### PR DESCRIPTION
**Pull request by PhilipAlbrecht**
fixes #7279

Goal:

Ticket(s): #

- [ ] Backport to LTR documentation is requested

### CODE REVIEW

File: docs/user_manual/introduction/qgis_gui.rst [hunk 1]
```diff
@@ -2488,36 +2494,35 @@ the view into 3D.
 
 The following tools are provided at the top of the 3D map view panel:
 
-* |pan| :sup:`Camera control`: moves the view, keeping the same angle
+* |pan| :sup:`Camera Control`: moves the view, keeping the same angle
```
Comment (DelazJ) [inline docs/user_manual/introduction/qgis_gui.rst:2497]: I'm not really fan of these capitalizations in the middle of a text but maybe, I'm french-biased...

**Review by DelazJ — COMMENTED** [review]
Can you remove changes from #7588 in this, please?
I'd prefer a less darker but more featured image for the 3D view. The project file we used here is one provided by Lutra guys some time ago (Luxembourg, it's named). Will try to upload a copy later today.

**Comment by DelazJ** [comment]
>  I'd prefer a less darker but more featured image for the 3D view. The project file we used here is one provided by Lutra guys some time ago (Luxembourg, it's named). Will try to upload a copy later today.

Found it: see https://docs.qgis.org/3.22/en/docs/user_manual/style_library/3d_symbols.html#application-example

File: docs/user_manual/introduction/qgis_gui.rst [hunk 2]
```diff
@@ -2488,36 +2494,35 @@ the view into 3D.
...
-  * |checkbox| :guilabel:`Smooth edges`
-  * |checkbox| :guilabel:`Export normals`
-  * |checkbox| :guilabel:`Export textures`
+ * :guilabel:`Scene name` and destination :guilabel:`Folder`
+ * :guilabel:`Terrain resolution`
+ * :guilabel:`Terrain texture resolution`
+ * :guilabel:`Model scale`
+ * |checkbox| :guilabel:`Smooth edges`
+ * |checkbox| :guilabel:`Export normals`
+ * |checkbox| :guilabel:`Export textures`
```
[the remark this answers is not in the document]
Comment (PhilipAlbrecht) [reply docs/user_manual/introduction/qgis_gui.rst:2521]: ok

**Comment by DelazJ** [comment]
@PhilipAlbrecht no, too heavy imho for the use we'd have. You can pick it at the link added above.

**Comment by PhilipAlbrecht** [comment]
Screenshot is updated

**Comment by DelazJ** [comment]
> Screenshot is updated

TBH I'd prefer a more featured screenshot. That screenshot does not really reflect what QGIS is able to do in 3D.
And if you consider that this screenshot is among those live fetched by the [code repo readme page](https://github.com/qgis/QGIS#features) from the docs to highlight QGIS sexiest features, I'm not sure we'll be doing great...

**Comment by PhilipAlbrecht** [comment]
Ok, I misunderstood that. I thought you meant the screenshot. But I also think that a screenshot with more content would be better. I have pasted the original screenshot again. I could not get more out of the data of the Lutraguys. Alternatively, I could try something with our own demo data from Düsseldorf. What do you think?

**Comment by DelazJ** [comment]
> Ok, I misunderstood that. I thought you meant the screenshot.

Yes, sorry. I meant the files. I should try to be more verbose in my reply 😄 
> Alternatively, I could try something with our own demo data from Düsseldorf. What do you think?

No problem. Try with what you are comfortable with, as long as it is good looking and showcases QGIS features. Thanks @PhilipAlbrecht

**Comment by DelazJ** [comment]
> I could not get more out of the data of the Lutraguys.

I confirm. Tried the project and lagged a lot, with 3d view getting blank (for no obvious reason). Even starting from a new view. I don't know if the project is the culprit or the 3D view has bugs...
If you don't have time to provide a new 3D screenshot soon, we can merge this in the meantime...

**Comment by PhilipAlbrecht** [comment]
> > I could not get more out of the data of the Lutraguys.
> 
> I confirm. Tried the project and lagged a lot, with 3d view getting blank (for no obvious reason). Even starting from a new view. I don't know if the project is the culprit or the 3D view has bugs... If you don't have time to provide a new 3D screenshot soon, we can merge this in the meantime...

I couldn't make a better sreenshot with our demodata. Think it's quite nice with the tress and the different buildings heights in it.
So lets go for this

**Pull request merged**